src/workload/mod.rs

```diff
@@ -761,10 +785,22 @@ impl Ratelimit {
     }
 
     pub fn next_ratelimit(&mut self) -> u64 {
-        if self.current + self.step <= self.end {
-            self.current += self.step;
+        let limit = self.limits[self.current_idx];
+        self.current_idx += 1;
+
+        if self.current_idx == self.limits.len() {
+            // If the test is to be looped or mirrored reset the pointer to the
+            // beginning of the vector and start again, else move back to the
+            // previous (final stable) value
+            if self.ramp_completion_action == RampCompletionAction::Loop
+                || self.ramp_completion_action == RampCompletionAction::Mirror
```

Comment:
I'm confused here. Shouldn't mirroring reverse our direction? This looks like both loop and mirror do the same thing?

Reply:
Mirrored doubles the index array and stores a reverse version as well so each cycle around the array is the ramp up and down. At the end of the cycle, both mirrored and loop just reset and run the cycle from the start again.